src/sync/mod.rs

```diff
@@ -1,11 +1,11 @@
 #![allow(clippy::redundant_pattern_matching)] // For derive(DeJson).
 
-use super::types::*;
 use crate::checksum;
 use crate::checksum::Checksum;
 use crate::dag;
 use crate::db;
 use crate::db::{Commit, Whence, DEFAULT_HEAD_NAME};
+use crate::embed::types::*;
```

Comment:
This creates a circular dependency between embed and sync. The way I imagined it was more like embed depends on sync (analogously to how embed depends on db).

I think we should try not to have circular deps in this repository, even though Rust allows, because a strict dag is so much easier to reason about.

Long story short: can we define the sync-related types the sync package, not embed?

I believe you can simply use them from embed either way.